flink/src/main/java/org/apache/iceberg/flink/sink/FlinkSink.java

```diff
@@ -140,13 +141,14 @@ private Builder forRowData(DataStream<RowData> newRowDataInput) {
                                             MapFunction<T, RowData> mapper,
                                             TypeInformation<RowData> outputType) {
       this.inputCreator = newUidPrefix -> {
+        // Input stream order is crucial for some situation(e.g. in cdc case), So we need to set the map opr
+        // parallelism as it's input to keep map opr chain to input, and ensure input stream will not be rebalanced.
```

Comment:
I agree with this change. Since map operator is part of the Flink sink, it is ok to set the parallelism.